### ISSUE TITLE
How to construct the test.json after download the hot3d_test?

### ISSUE THREAD

**Issue by EAST-J** [Biscue5/EgoScaler#8]
Hi! Thanks for the excellent work and for the open-source release. 
I've downloaded the HOT3D test data and was trying to evaluate the released checkpoint's performance, but I couldn't find a test.json file in the datasets. Do you have any instructions on how to create this file?

**Comment by EAST-J** [comment]
I successfully used format_data.py to construct the dataset.
However, I encountered several issues in the dataset loading. In particular, the file ```egoscaler/models/pointllm/dataset.py``` triggers errors related to a missing function ```self.get_padding```.  I temporarily implemented the function as follows:
```
  def get_padding(self, token_ids, max_len, pad_token_id=0):
      """
      Pads or truncates the token_ids to length max_len.
      
      Args:
          token_ids (list[int]): tokenized ids
          max_len (int): desired final length
          pad_token_id (int): padding id from tokenizer

      Returns:
          padded (list[int]): padded token ids
          mask (list[int]): attention mask, 1 for real tokens, 0 for padding
      """
      # --- truncate if too long ---
      if len(token_ids) > max_len:
          token_ids = token_ids[:max_len]

      # --- compute mask (before padding) ---
      mask = [1] * len(token_ids)

      # --- apply padding ---
      pad_len = max_len - len(token_ids)
      if pad_len > 0:
          token_ids = token_ids + [pad_token_id] * pad_len
          mask = mask + [0] * pad_len

      return token_ids, mask
``` 
However, I found that ```self.tokenizer.pad_token_id``` is ```None```, and it fails to convert to the tensors. Could you provide guidance on how to correctly configure the tokenizer or dataset?

**Comment by EAST-J** [comment]
Thanks for your reply, and I still have some followed questions:
- My evaluation results are slightly higher than the reported baseline (ADE: 0.48, FDE: 0.79, GD: 0.99$). Given the potential difference in the test set or environment, is this deviation considered reasonable and expected?
- The current implementation appears to treat the full trajectory (position and rotation) uniformly for ADE and FDE:
```
ade: float = average_displacement_error(gen_traj[None,:,:], gt_traj[None,:,:])
fde: float = final_displacement_error(gen_traj[None,:,:], gt_traj[None,:,:])
gd: float = anglar_distance(gen_traj[None, :, :], gt_traj[None, : ,:])
```
Should the calculation for position (ADE/FDE) and rotation (GD) be separated to correctly measure the $xyz$ coordinates and rotation components, respectively?
```
ade: float = average_displacement_error(gen_traj[:, :3], gt_traj[:,:3])  # Only Position (XYZ)
fde: float = final_displacement_error(gen_traj[:, :3], gt_traj[:, :3])  # Only Position (XYZ)
gd: float = anglar_distance(gen_traj[:, 3:], gt_traj[: , 3:])          # Only Rotation (R P Y / Quat)
```
- The separator token, SEP_TOKEN <sep>, appears to be missing from the tokenizer's vocabulary. Should this token be added as a special token during initialization?
- The function ```dataset.detokenize_traj(gen_traj, num_bins=args.num_bins, max_abs=m_abs)``` is missing. I currently use a ```str_to_float``` function for conversion. Could you please share the original or intended implementation of this detokenization function?
- For the reported MiniGPT-3D results, was the model trained using LoRA (as detailed in the original publication) or through full parameter fine-tuning?

**Comment by Biscue5** [comment]
The numbers you obtained are considered within a reasonable range. You may want to add the `<sep>` token explicitly if it’s missing. I added the missing function and updated the metric implementation accordingly, so please pull the latest changes. MiniGPT-3D uses full fine-tuning on the LLM part as well.
